### ISSUE TITLE
md-input broken in IE11

### ISSUE THREAD

**Issue by msegers** [angular/material2#730]
#### Bug, feature request, or proposal:
Bug

#### What is the expected behavior?
All features of md-input are working as expected, label, hint etc.

#### What is the current behavior?
no placeholder
no label
no hint
no properties are linked through

#### What are the steps to reproduce?
http://plnkr.co/edit/5vCYmKJ0G0On7uFAb9P2?p=preview

I"m trying to create a plunker but it just crashes when I try to use the template plunk.
Is there any demo app with latest dependencies? As far as I can see the material2 demo on firebaseapp is not up to date (Can't type there on IE which is like alpha 3/4)

But basically try it on IE11 Win 10

#### What is the use-case or motivation for changing an existing behavior?
I want IE users to type into md-input fields, not normal input fields

#### Which versions of Angular, Material, OS, browsers are affected?
RC 1, Material 5-2, Windows 10, IE11
Works fine on Edge

#### Is there anything else we should know?
My windows 10 intstall is a VirtualBox instance

**Comment by msegers** [comment]
<img width="878" alt="screen shot 2016-06-21 at 10 20 42" src="https://cloud.githubusercontent.com/assets/3832687/16222745/06673210-379a-11e6-8047-ed072257828b.png">

This is the error that I can see on when trying to load the default plunker where md-button is replaced with md-input and an md-hint, only applied a placeholder. Works fine in other browsers.

**Comment by msegers** [comment]
FYI, I can't seem to find any working Plunkers in IE11, nor without Material nor with

**Comment by hansl** [comment]
I don't know where you found this plnkr, but you are missing a polyfill. The CLI (and Material) normally uses Core-JS, which you can use by adding this to your index.html:

```html
    <script src="https://cdnjs.cloudflare.com/ajax/libs/core-js/2.4.0/core.min.js"></script>
```

@jelbourn: What's our official Material2 plnkr so I can update those templates?

Closing this as it's not an issue with Material.

**Comment by jelbourn** [comment]
@hansl: @DevVersion created the plunkr. You can fork the template and then point our readme/issue template to it.

**Comment by msegers** [comment]
@hansl Sorry to bother you, but you might get me further. Is this something which is broken right now (and in this case not a problem with material). Or is it something that should actually work, and that I need to sort my polyfills. I'm using the angular2-webpack-starter project and that includes the core-js package.

I'm loading that into the polyfills. Also I've added the script tag you posted just now to see if that changes anything but I still have these errors going on.

So in short, should I check my polyfills or just wait for an update in core-js?

**Comment by hansl** [comment]
You should check your polyfills. core-js already has this.

**Comment by DevVersion** [comment]
@hansl I've updated the Plunker Template to use corejs now. Also dropped the dependency on Reflect-Metadata,

**Comment by msegers** [comment]
I'll check the Plunker, I still have the issue despite having core-js so it must be my own implementation causing issues. maybe mixing ngControl & ngModel or smth.

**Comment by msegers** [comment]
@hansl Just fyi, I guess you might already know. I found a way to debug IE11 properly. My issue occurs when you have a form with an md-input using type=date on IE11.

IE11 doesn't support date and for some reason it has issues, therefore I solved it in my case by optionally showing a regular text input on browsers which don't support the date input type.

Hope this info helps anything in the future, sorry to bother.

**Comment by msegers** [comment]
Oh: here is finally a usefull plnkr http://plnkr.co/edit/mraL1tJHVL2sP8BSU85d?p=preview
